project-api/project-api-test/src/test/java/io/ballerina/projects/test/PackageResolutionLocalTests.java

```diff
@@ -0,0 +1,359 @@
+                readFileAsString(projectDirPath.resolve(RESOURCE_DIR_NAME).resolve("Dependencies.toml")));
+    }
+
+    @Test(description = " Local repository packages (published, but patched with the same version " +
+            "with different indirect deps) as direct dependencies")
+    public void testCase0007(ITestContext ctx) throws IOException {
+        /*
+            project_0007 ---> package_e:0.0.1 (central) ---> package_f:0.0.1 (central)
+
+            project_0007 ---> package_e:0.0.1 (local) ---> package_f:0.0.1 (central)
+                              package_e:0.0.1 (local) ---> package_g:0.0.1 (central)
+
+            Package_e resolved from local repo as expected.
+         */
+
+        // Note that we are using a similar project to the previous test scenario project_0004
+        Path projectDirPath = PROJECTS_DIRECTORY.resolve("project_0007");
+        ctx.getCurrentXmlTest().addParameter("packagePath", String.valueOf(projectDirPath));
+
+        // Push package_f:0.0.1 to central
+        cacheDependencyToCentralRepository(PACKAGES_DIRECTORY.resolve("package_f_0_0_1_central"),
+                projectEnvironmentBuilder);
+
+        // Push package_e:0.0.1 to central
+        cacheDependencyToCentralRepository(PACKAGES_DIRECTORY.resolve("package_e_0_0_1_central"),
+                projectEnvironmentBuilder);
+
+        // Build project and assert deps toml
+        BuildProject buildProject = BuildProject.load(projectEnvironmentBuilder, projectDirPath);
+        buildProject.save();
+        failIfDiagnosticsExists(buildProject);
+
+        Assert.assertEquals(readFileAsString(projectDirPath.resolve(DEPENDENCIES_TOML)),
+                readFileAsString(projectDirPath.resolve(RESOURCE_DIR_NAME).resolve("Dependencies.toml")));
+
+        // Delete it manually since aftermethod only works for previous project
+        deleteDependenciesToml(projectDirPath);
+
+        // Change the deps in toml
+        projectDirPath = PROJECTS_DIRECTORY.resolve("project_0007_v2");
+        ctx.getCurrentXmlTest().addParameter("packagePath", String.valueOf(projectDirPath));
+
+        // Push package_g:0.0.1 to central
+        cacheDependencyToLocalRepository(PACKAGES_DIRECTORY.resolve("package_g_0_0_1_central"),
+                projectEnvironmentBuilder);
+
+        // Push package_e:0.0.1 to local
+        cacheDependencyToLocalRepository(PACKAGES_DIRECTORY.resolve("package_e_0_0_1"), projectEnvironmentBuilder);
+
+        // Build project and assert deps toml
+        buildProject = BuildProject.load(projectEnvironmentBuilder, projectDirPath);
+        buildProject.save();
+        failIfDiagnosticsExists(buildProject);
+
+        Assert.assertEquals(readFileAsString(projectDirPath.resolve(DEPENDENCIES_TOML)),
+                readFileAsString(projectDirPath.resolve(RESOURCE_DIR_NAME).resolve("Dependencies.toml")));
+    }
+
+    @AfterMethod
+    private void deleteBuildFile(ITestContext ctx) throws IOException {
```

Comment:
Shall we change this after the method, so we can use it for all the test methods?
Maybe we can pass a boolean to identify whether we need to delete the `Dependencies.toml`.

Reply:
Merged to one method